nixos/modules/services/web-apps/opencloud.nix

```diff
@@ -0,0 +1,212 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib) types;
+  cfg = config.services.opencloud;
+  defaultUser = "opencloud";
+  defaultGroup = defaultUser;
+in
+{
+  options = {
+    services.opencloud = {
+      enable = lib.mkEnableOption "OpenCloud";
+
+      package = lib.mkPackageOption pkgs "opencloud" { };
+      webPackage = lib.mkPackageOption pkgs [ "opencloud" "web" ] { };
+      idpWebPackage = lib.mkPackageOption pkgs [ "opencloud" "idp-web" ] { };
+
+      configDir = lib.mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/lib/opencloud/config";
+        description = ''
+          Path to directory containing OpenCloud config file.
+
+          Example config can be generated by `opencloud init --config-path fileName --admin-password "adminPass"`.
+          Add `--insecure true` if SSL certificates are generated and managed externally (e.g. using OpenCloud behind a reverse proxy).
+
+          Note: This directory must contain at least a `opencloud.yaml`. Ensure
+          [user](#opt-services.opencloud.user) has read/write access to it. In some
+          circumstances you may need to add additional OpenCloud configuration files (e.g.,
```

Comment:
In what circumstances? We should phrase it like: additional config files like proxy.yml have to be placed in this directory, too.